Replication Project_Final.Rmd

```diff
@@ -0,0 +1,858 @@
+  select(muncode, hom.total)
+```
+
+**CREATING THE MAIN DATAFRAME OF DATA FROM STANDARD CENSUS, PLUS HOMICIDES**
+**Adding Dummy Variable for State of Mexico**
+```{r, warning=FALSE, error=FALSE}
+#join all new variables to creat the main data frame with control and dependent variables
+main <- left_join(mun.total, area, by = "muncode")
+main <- left_join(main, docs, by = "muncode")
+main <- left_join(main, fems, by = "muncode")
+main <- left_join(main, young, by = "muncode")
+main <- left_join(main, homicides, by = "muncode")
+
+#adding dummy variable for state of Mexico and creating name variable in order to join with agricultural variables dataframe later
+main <- main %>% 
+  mutate(dummy.SOM = as.numeric(muncode %in% 15001:15125)) %>%
+  mutate(name = paste(mun_name, state_name, sep = ", "))
+```
+
+**Agricultural Data**
+```{r, warning=FALSE, error=FALSE}
+#load all the files
+prod_units <- read.csv(file="data/censo_agricola_1991_produnits.csv", header=FALSE, skip = 8, stringsAsFactors = FALSE, encoding = "UTF-8")
+prod_units <- tbl_df(prod_units)
+land <- read.csv(file="data/censo_agricola_1991_land_area.csv", header=FALSE, skip = 8, stringsAsFactors = FALSE, encoding = "UTF-8")
+land <- tbl_df(land)
+subs <- read.csv("data/censo_agricola_1991_autoconsumo.csv", header = FALSE, skip = 11, stringsAsFactors = FALSE, encoding = "UTF-8")
+subs <- tbl_df(subs)
+cattle <- read.csv("data/censo_agricola_1991_cattle.csv", header = FALSE, skip = 10, stringsAsFactors = FALSE, encoding = "UTF-8")
+cattle <- tbl_df(cattle)
+coffee <- read.csv("data/censo_agricola_1991_coffee.csv", header = FALSE, skip = 1069, nrows = 1262, stringsAsFactors = FALSE, encoding = "UTF-8")
+coffee <- tbl_df(coffee)
+corn <- read.csv("data/censo_agricola_1991_corn.csv", header = FALSE, skip = 6187, nrows = 2426, stringsAsFactors = FALSE, encoding = "UTF-8")
+corn <- tbl_df(corn)
+
+#function that corrects municipality names and creates a new variable with pasted municipality name and state
+naming <- function(df) {
+  df$V3 <- gsub("[[:space:]]", "", df$V3)
+  df$V4 <- gsub("[[:space:]]", "", df$V4)
+  df$V5 <- gsub("[[:space:]]", "", df$V5)
+  df$V6 <- gsub("[[:space:]]", "", df$V6)
+  df$V7 <- gsub("[[:space:]]", "", df$V7)
+  
+  df1 <- df %>%
+    select(state = V1, mun = V2, V3, V4, V5, V6, V7) %>%
+    mutate(state = as.character(state), mun = as.character(mun), 
+           V3 = as.numeric(V3), V4 = as.numeric(V4),
+           V5 = as.numeric(V5), V6 = as.numeric(V6)  , V7 = as.numeric(V7))           
+  
+  df1$state[df1$state == ""] <- NA
+  df1$mun[df1$mun == ""] <- NA
+  df1$mun <- gsub("\x84", "N",df1$mun) 
+  df1$state <- na.locf(df1$state)
+  
+  df1 <- df1 %>%
+    separate(mun, into = c("mun", "art"), sep = ",", extra = "merge") %>%
+    trim(df1$art) %>% 
+    mutate(mun = ifelse(!is.na(art), paste(art, mun, sep = " "), mun)) %>%
+    filter(!is.na(mun)) %>%
+    mutate(name = paste(mun, state, sep = ", ")) %>%
+    select(-art)   
+}
```

Comment:
this function is not well documented; what are the inputs and what are the outputs